frontend/web/components/segments/SegmentRow/SegmentRow.tsx

```diff
@@ -0,0 +1,160 @@
+import { FC } from 'react'
+import { RouterChildContext } from 'react-router'
+
+import { useHasPermission } from 'common/providers/Permission'
+
+import Utils from 'common/utils/utils'
+import Icon from 'components/Icon'
+import ConfirmRemoveSegment from 'components/modals/ConfirmRemoveSegment'
+
+import { Segment } from 'common/types/responses'
+import { MutationDefinition } from '@reduxjs/toolkit/dist/query/endpointDefinitions'
+import { MutationTrigger } from '@reduxjs/toolkit/dist/query/react/buildHooks'
+import SegmentAction from './components/SegmentAction'
+import ConfirmCloneSegment from 'components/modals/ConfirmCloneSegment'
+import { useCloneSegmentMutation } from 'common/services/useSegment'
+
+interface SegmentRowProps {
+  segment: Segment
+  index: number
+  projectId: string
+  router: RouterChildContext['router']
+  removeSegment: MutationTrigger<
```

Comment:
Awesome! I never knew about this.

Would 

```
{
        projectId: number | string
        id: number
      }
```

Simply be Req['removeSegment']